### ISSUE TITLE
[1.0.2] Crash in identify with analytics 3.0.4

### ISSUE THREAD

**Issue by ldiqual** [segment-integrations/analytics-ios-integration-mixpanel#5]
```
2016-01-07 14:29:39.296 App[50163:5972653] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: '-[__NSCFDictionary setObject:forKey:]: mutating method sent to immutable object'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010697ce65 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x000000010a656deb objc_exception_throw + 48
	2   CoreFoundation                      0x000000010697cd9d +[NSException raise:format:] + 205
	3   CoreFoundation                      0x00000001068a9863 -[__NSCFDictionary setObject:forKey:] + 99
	4   CoreFoundation                      0x0000000106898a8d -[NSMutableDictionary addEntriesFromDictionary:] + 301
	5   Analytics                           0x0000000105fa1b7d __35-[SEGSegmentIntegration addTraits:]_block_invoke + 77
	6   Analytics                           0x0000000105f99165 seg_dispatch_specific + 117
	7   Analytics                           0x0000000105f99212 seg_dispatch_specific_async + 82
	8   Analytics                           0x0000000105fa0f6f -[SEGSegmentIntegration dispatchBackground:] + 63
	9   Analytics                           0x0000000105fa1ae5 -[SEGSegmentIntegration addTraits:] + 181
	10  Analytics                           0x0000000105fa2065 __34-[SEGSegmentIntegration identify:]_block_invoke + 165
	11  libdispatch.dylib                   0x000000010b130e5d _dispatch_call_block_and_release + 12
	12  libdispatch.dylib                   0x000000010b15149b _dispatch_client_callout + 8
	13  libdispatch.dylib                   0x000000010b1378ec _dispatch_queue_drain + 2215
	14  libdispatch.dylib                   0x000000010b136e0d _dispatch_queue_invoke + 601
	15  libdispatch.dylib                   0x000000010b139a56 _dispatch_root_queue_drain + 1420
	16  libdispatch.dylib                   0x000000010b1394c5 _dispatch_worker_thread3 + 111
	17  libsystem_pthread.dylib             0x000000010b49f4f2 _pthread_wqthread + 1129
	18  libsystem_pthread.dylib             0x000000010b49d375 start_wqthread + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

![image](https://cloud.githubusercontent.com/assets/941453/12184721/e9fee508-b54b-11e5-9120-7abb2e6fb799.png)
![image](https://cloud.githubusercontent.com/assets/941453/12184734/0392656c-b54c-11e5-8d66-07960844214a.png)

**Comment by ldiqual** [comment]
FYI: reverting analytics to 3.0.3 fixes the issue.

**Comment by f2prateek** [comment]
Van you share your code?

**Comment by f2prateek** [comment]
I'm guessing it is related to https://github.com/segmentio/analytics-ios/pull/488

**Comment by f2prateek** [comment]
Also please open this against the right repo https://github.com/segmentio/analytics-ios

**Comment by ldiqual** [comment]
Seems fixed in analytics-ios 3.0.6. Thanks!
